packages/eas-cli/src/submit/android/ServiceAccountSource.ts

```diff
@@ -30,66 +31,112 @@ interface ServiceAccountPromptSource extends ServiceAccountSourceBase {
   sourceType: ServiceAccountSourceType.prompt;
 }
 
-interface ServiceAccountDetectSource extends ServiceAccountSourceBase {
-  sourceType: ServiceAccountSourceType.detect;
+interface ServiceAccountCredentialsServiceSource extends ServiceAccountSourceBase {
+  sourceType: ServiceAccountSourceType.credentialsService;
 }
 
+export type ServiceAccountKeyResult = {
+  result: ServiceAccountKeyFile | ServiceAccountKeyFromExpoServers;
+  summary: ServiceAccountKeySummary;
+};
+
+type ServiceAccountKeySummary = {
+  source: 'local' | 'Expo servers';
+  path?: string;
+  email: string;
+};
+
+type ServiceAccountKeyFile = {
+  googleServiceAccountKeyJson: string;
+};
+
+type ServiceAccountKeyFromExpoServers = {
+  googleServiceAccountKeyId: string;
+};
+
 export type ServiceAccountSource =
   | ServiceAccountPathSource
   | ServiceAccountPromptSource
-  | ServiceAccountDetectSource;
+  | ServiceAccountCredentialsServiceSource;
+
+export async function getServiceAccountKeyResultAsync(
+  ctx: SubmissionContext<Platform.ANDROID>,
+  source: ServiceAccountSource,
+  androidApplicationIdentifier: string
+): Promise<ServiceAccountKeyResult> {
+  if (source.sourceType !== ServiceAccountSourceType.credentialsService) {
+    const serviceAccountKeyPath = await getServiceAccountKeyPathAsync(source);
+    const serviceAccountKey = readAndValidateServiceAccountKey(serviceAccountKeyPath);
+    return {
+      result: { googleServiceAccountKeyJson: await fs.readFile(serviceAccountKeyPath, 'utf-8') },
+      summary: {
+        source: 'local',
+        path: serviceAccountKeyPath,
+        email: serviceAccountKey.client_email,
+      },
+    };
+  }
+  return await getServiceAccountFromCredentialsServiceAsync(
+    ctx,
+    source,
+    androidApplicationIdentifier
+  );
+}
 
-export async function getServiceAccountAsync(source: ServiceAccountSource): Promise<string> {
+export async function getServiceAccountKeyPathAsync(source: ServiceAccountSource): Promise<string> {
   switch (source.sourceType) {
     case ServiceAccountSourceType.path:
       return await handlePathSourceAsync(source);
     case ServiceAccountSourceType.prompt:
       return await handlePromptSourceAsync(source);
-    case ServiceAccountSourceType.detect:
-      return await handleDetectSourceAsync(source);
   }
+  throw new Error(`ServiceAccountSource ${source} does not return a path.`);
+}
+
+export async function getServiceAccountFromCredentialsServiceAsync(
+  ctx: SubmissionContext<Platform.ANDROID>,
+  _source: ServiceAccountCredentialsServiceSource,
```

Comment:
```suggestion
```
I think it's not necessary to add a parameter "for the future". It may never come :D